v2/periphery/LockupLinearBatchStreamCreator.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.13;
+
+import { IAllowanceTransfer } from "@permit2/interfaces/IAllowanceTransfer.sol";
+import { IPRBProxy } from "@prb/proxy/interfaces/IPRBProxy.sol";
+import { IPRBProxyRegistry } from "@prb/proxy/interfaces/IPRBProxyRegistry.sol";
+import { ISablierV2LockupLinear } from "@sablier/v2-core/interfaces/ISablierV2LockupLinear.sol";
+import { ud60x18 } from "@sablier/v2-core/types/Math.sol";
+import { IERC20 } from "@sablier/v2-core/types/Tokens.sol";
+import { ISablierV2ProxyTarget } from "@sablier/v2-periphery/interfaces/ISablierV2ProxyTarget.sol";
+import { Batch, Broker, LockupLinear, Permit2Params } from "@sablier/v2-periphery/types/DataTypes.sol";
+import { BatchBuilder } from "@sablier/v2-periphery-test/utils/BatchBuilder.sol";
+
+contract LockupLinearBatchStreamCreator {
+    IERC20 public constant DAI = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    IAllowanceTransfer public constant PERMIT2 = IAllowanceTransfer(0x000000000022D473030F116dDEE9F6B43aC78BA3);
+    IPRBProxyRegistry public constant PROXY_REGISTRY = IPRBProxyRegistry(0x33e200B5fb5e0C57d370d5202c26A35d07A46B98);
+    ISablierV2LockupLinear public immutable sablierLockupLinear;
+    ISablierV2ProxyTarget public immutable sablierProxyTarget;
+
+    constructor(ISablierV2LockupLinear sablierLockupLinear_, ISablierV2ProxyTarget sablierProxyTarget_) {
+        sablierLockupLinear = sablierLockupLinear_;
+        sablierProxyTarget = sablierProxyTarget_;
+    }
+
+    function batchCreateLockupLinearStream(
+        uint256 perStreamAmount,
+        uint256 batchSize,
+        bytes memory signature
```

Comment:
It would be quiet to repetitive to have `permit2Params.permit2Signature`

Reply:
It doesn't matter - this is an example. Clarity is more important - a signature can mean all sorts of different things in Ethereum.

Reply:
Actually - we would never get to write `permit2Params.permit2Signature`, because the `Permit2Params` struct defines `signature`. We would write this:

```solidity
permit2Params.signature = permit2Signature;
```

Which is fine!

Reply:
Resolved in the last commit